Assets/Scripts/Controllers/TileSpriteController.cs

```diff
@@ -56,11 +56,16 @@ protected override void OnCreated(Tile tile)
         sr.sprite = SpriteManager.current.GetSprite("Tile", "Empty");
         sr.sortingLayerName = "Tiles";
 
-        OnChanged(tile);
+        OnChanged(tile,true);
+    }
+
+    protected override void OnChanged(Tile obj)
+    {
+        OnChanged(obj, false);
     }
 
     // This function should be called automatically whenever a tile's data gets changed.
-    protected override void OnChanged(Tile tile)
+    protected void OnChanged(Tile tile, bool forceTile)
```

Comment:
The force tile is used for when first creating a tile to ensure it is in the proper state. I will agree that it doesn't do anything at the moment. The tile sprite is actually changed every time it is walked on at the moment, I had in mind a "force" state to only change if the conditions changed, or if the tile was created. I can continue with those improvements prior to implementation if you think it best.